### PR TITLE
Python 3 Support

### PR DESCRIPTION
**Pull request by pipermerriam**
### What was wrong?

* Travis CI is not enabled
* There was a `print` statement that isn't python 3 compatable.
* The `market.se` contract needed to be updated to the new extern format (see https://github.com/ethereum/serpent/issues/85)
* The package didn't specify anything about what versions of python it is compatable with.
* No python3 support.
* The README was very sparse in general information about the package.

### How was it fixed?

* Added a basic `.travis.yml` and `tox.ini` file.
* Converted the print statement to be python3 compatable.
* Redid the `extern` statements to be correct.
* Added classifiers for python 2.7/3.4/3.5
* Added info to the README on how to run the tests.

#### Cute animal picture

f55a1348d24098a680f5f4ff3838dca5

### CODE REVIEW

**Comment by pipermerriam** [comment]
This PR should resolve the following issues.

* https://github.com/ethereum/serpent/issues/94 - Python 3 compatability.
* #85 - compile error on the `market.se` example contract.

**Comment by pipermerriam** [comment]
@vbuterin could you take a look at this?  Potentially enable travis-ci on this repo and let me know what's needed to get this merged.

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    -r{toxinidir}/requirements-dev.txt
+
+[testenv:py27]
+basepython=python2.7
+
+[testenv:py34]
+basepython=python3.4
+
+[testenv:py35]
+basepython=python3.5
```
Comment (pjdelport) [inline tox.ini]: These `basepython` declarations shouldn't be needed anymore for non-ancient versions of Tox.

File: setup.py [hunk 2]
```diff
@@ -43,5 +43,13 @@
...
-    }
-    ),
+    },
+    classifiers=[
+        'Intended Audience :: Developers',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.3',
+        'Programming Language :: Python :: 3.4',
```
Comment (pjdelport) [inline setup.py]: Should this be adjusted to 3.4 + 3.5 rather, to reflect the support better?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    py27,
+    py34,
+    py35
+
+[flake8]
+max-line-length= 100
+exclude= tests/*
+
+[testenv]
+commands=py.test --tb native {posargs:tests}
```
Comment (pjdelport) [inline tox.ini]: The `tests` path here isn't correct for this repo: it should probably either be `.` or `examples/cyberdyne/test`?

File: setup.py (hunk 2 above)
Comment (pipermerriam) [inline setup.py]: Thank you, copy/paste error.

File: tox.ini (hunk 3 above)
Comment (pipermerriam) [inline tox.ini]: `py.test` finds all of the appropriate test files using this command.

File: tox.ini (hunk 1 above)
Comment (pipermerriam) [inline tox.ini]: Neat, I didn't know that. fixed.

**Comment by pipermerriam** [comment]
@tgerring I was told that you might be able to enable travis-ci for this repository?  Also, looking for someone who has merge permissions.  I got the 👍 from @vbuterin over skype.

File: tox.ini (hunk 3 above)
Comment (pjdelport) [inline tox.ini]: Hmm, that's weird. I get `ERROR: file not found: tests`, as I would expect. There's no `tests` directory in the repo: what is it actually finding?

Comment (pipermerriam) [inline tox.ini]: It recursively searches the project for something like `**/tests/test_*.py`.  That is probably not the exact glob pattern but you get the idea.

Comment (pipermerriam) [inline tox.ini]: Actually, you're right.  I was actually in the wrong window and running tests on a different repo :)

Fixing for real this time.

**Comment by tgerring** [comment]
@pipermerriam The ethereum/serpent repo has been activated on Travis

File: tox.ini (hunk 3 above)
Comment (pjdelport) [inline tox.ini]: Phew, okay, good to know I'm not losing sanity. :)

**Comment by pipermerriam** [comment]
ok, looks like the test failures are related to not having python3 in `pyethereum`.  I'm going to deal with this on the pyethereum side and then we'll get this merged.

**Comment by pipermerriam** [comment]
Just noting that the tests for this pass when run against the pyethereum branch with python 3 support.  See #103

**Comment by pipermerriam** [comment]
Merging this early with the `pyethereum` dependency pinned to the unreleased develop version for the time being so that other tools can get moving forward.  Will wait to release until an official `pyethereum` version has been released.

**Pull request merged**